### ISSUE TITLE
build fails ( Java 14.0.1 )

### ISSUE THREAD

**Issue by grimace** [ArcadeData/arcadedb#16]
## Expected behavior
build should complete successfully

## Actual behavior
failure building engine

## Steps to reproduce
clone to local project folder
cd arcadedb
mvnw install

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project arcadedb-engine: Compilation failure: Compilation failure: 
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/database/async/DatabaseAsyncExecutorImpl.java:[376,34] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/serializer/BinarySerializer.java:[51,58] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/engine/Bucket.java:[84,33] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/engine/Bucket.java:[88,34] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/engine/Bucket.java:[192,19] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/engine/Bucket.java:[374,42] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/engine/Bucket.java:[452,46] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/database/async/DatabaseAsyncExecutor.java:[75,21] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/engine/BucketIterator.java:[32,49] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/engine/BucketIterator.java:[40,3] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/engine/BucketIterator.java:[126,10] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/schema/EmbeddedSchema.java:[335,23] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/graph/ImmutableVertex.java:[61,11] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/query/sql/function/misc/SQLFunctionEncode.java:[57,13] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] [User Path]/projects/arcadedb/engine/src/main/java/com/arcadedb/graph/ImmutableEdge.java:[61,11] reference to Record is ambiguous
[ERROR]   both interface com.arcadedb.database.Record in com.arcadedb.database and class java.lang.Record in java.lang match
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :arcadedb-engine
```

**Comment by lvca** [comment]
We used Java11 to compile it. We should definitely check with the more recent compilers. Thanks for the report.

**Comment by lvca** [comment]
Fixed by commit 5cdd167.

**Comment by lvca** [comment]
Still some module needs a fix...

**Comment by lvca** [comment]
Fixed by commit 15eae32.

**Comment by lvca** [comment]
Even if the build works, Gremlin and Cypher are not compatible with JDK16+ because of a bug with Gradle. Issue: https://github.com/ArcadeData/arcadedb/issues/17.
